test/new.jl

```diff
@@ -0,0 +1,1817 @@
+            @test pkg.version == v"0.5.1"
+            @test isdir(pkg.source)
+            # - We also check the interaction between the previously intalled version.
+            @test pkg.source != source053
+        end
+        # Now a few more versions:
+        Pkg.add(Pkg.PackageSpec(;name="Example", version="0.5.0"))
+        Pkg.add(Pkg.PackageSpec(;name="Example"))
+        Pkg.add(Pkg.PackageSpec(;name="Example", version="0.3.0"))
+        Pkg.add(Pkg.PackageSpec(;name="Example", version="0.3.3"))
+        # With similar checks
+        Pkg.dependencies(exuuid) do pkg
+            @test pkg.version == v"0.3.3"
+            @test isdir(pkg.source)
+        end
+        # Now we try adding a second dependency.
+        # We repeat the same class of tests.
+        Pkg.add(Pkg.PackageSpec(;name="JSON", version="0.18.0"))
+        sourcej018 = nothing
+        Pkg.dependencies(json_uuid) do pkg
+            @test pkg.version == v"0.18.0"
+            @test isdir(pkg.source)
+        end
+        Pkg.add(Pkg.PackageSpec(;name="JSON", version="0.20.0"))
+        Pkg.dependencies(json_uuid) do pkg
+            @test isdir(pkg.source)
+            @test pkg.source != sourcej018
+        end
+        # Now check packages which track repos instead of registered versions
+        Pkg.add(Pkg.PackageSpec(;url="https://github.com/JuliaLang/Example.jl", rev="v0.5.3"))
+        Pkg.dependencies(exuuid) do pkg
+            @test !pkg.is_tracking_registry
+            @test isdir(pkg.source)
+            @test isdir(pkg.git_cache)
+        end
+        Pkg.add(Pkg.PackageSpec(;name="Example", rev="master"))
+        Pkg.dependencies(exuuid) do pkg
+            @test !pkg.is_tracking_registry
+            @test isdir(pkg.source)
+            @test isdir(pkg.git_cache)
+        end
+        # Also check that unregistered packages are installed properly.
+        Pkg.add(Pkg.PackageSpec(;url="https://github.com/00vareladavid/Unregistered.jl"))
+        Pkg.dependencies(unregistered_uuid) do pkg
+            @test isdir(pkg.source)
+            @test isdir(pkg.git_cache)
+        end
+        # Check `develop`
+        Pkg.develop(Pkg.PackageSpec(;name="Example"))
+        Pkg.dependencies(exuuid) do pkg
+            @test isdir(pkg.source) # TODO check for full git clone, have to implement saving original URL first
+        end
+        Pkg.develop(Pkg.PackageSpec(;name="JSON"))
+        Pkg.dependencies(json_uuid) do pkg
+            @test isdir(pkg.source) # TODO check for full git clone, have to implement saving original URL first
+        end
+        # Check that the original installation was undisturbed.
+        regs = Pkg.Registry.status(;as_api=true)
+        @test length(regs) == 1
+        reg = regs[1]
+        @test reg.name == "General"
+        @test reg.uuid == general_uuid
+        @test mtime(source053) == source053_time
+        # Now we clean up so that `isolate` can reuse the loaded depot properly
```

Comment:
I don't really understand the "loaded depot" terminology. The point is that the depot does not get reset, right? There is no need to "load" it initially with what we are going to use. Whatever ends up there ends up there.